scipy/stats/_stats_py.py

```diff
@@ -1089,6 +1090,35 @@ def moment(a, order=1, axis=0, nan_policy='propagate', *, center=None):
         return _moment(a, order, axis, mean=center)
 
 
+def _demean(a, mean, axis, *, xp, precision_warning=True):
+    # subtracts `mean` from `a` and returns the result,
+    # warning if there is catastrophic cancellation. `mean`
+    # must be the mean of `a` along axis with `keepdims=True`.
+    # Used in e.g. `_moment`, `_zscore`, `_xp_var`. See gh-15905.
+    a_zero_mean = a - mean
+
+    if xp_size(a_zero_mean) == 0:
+        return a_zero_mean
+
+    eps = xp.finfo(mean.dtype).eps * 10
+
+    with np.errstate(divide='ignore', invalid='ignore'):
+        rel_diff = xp.max(xp.abs(a_zero_mean), axis=axis,
+                          keepdims=True) / xp.abs(mean)
+    with np.errstate(invalid='ignore'):
+        precision_loss = xp.any(rel_diff < eps)
+    n = (xp_size(a) if axis is None
+         # compact way to deal with axis tuples or ints
+         else np.prod(np.asarray(a.shape)[np.asarray(axis)]))
```

Comment:
It looks like this line was missed during conversion?
```suggestion
         else xp.prod(xp.asarray(a.shape)[xp.asarray(axis)]))
```

Reply:
optional, but when it spans multiple lines I would find a standard `if`-`else` more readable than the ternary operator

Reply:
Yeah, thanks. Anything else to fix at the same time?

Reply:
Er, no, actually. We're just making use of the NumPy functions to work with `axis` because they are convenient here. Otherwise we have three separate cases to handle: `None`, integer, and sequence.

Reply:
I am happy to leave it but would you be able to make the comment slightly more verbose. It's just this jumps out when everything else is `xp`